File: parish360-core/src/main/java/org/parish360/core/family/dto/MiscellaneousInfo.java
package org.parish360.core.family.dto;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class MiscellaneousInfo {
    private String id;
    private Instant createdAt;
    private String createdBy;
    private Instant updatedAt;
    private String updatedBy;

    @NotNull(message = "comment by is not specified")
    private String commentedBy;

    @NotNull(message = "comment is not specified")
    private String comment;

    private String familyId;
}

File: parish360_mobile/lib/features/families/domain/entities/miscellaneous_info.dart
class MiscellaneousInfo {
  final String? id;
  final String? createdBy;
  final DateTime? createdAt;
  final String? updatedBy;
  final DateTime? updatedAt;
  final String? commentedBy;
  final String? comment;

  MiscellaneousInfo({
    this.id,
    this.createdBy,
    this.createdAt,
    this.updatedBy,
    this.updatedAt,
    this.commentedBy,
    this.comment,
  });

  factory MiscellaneousInfo.fromJson(Map<String, dynamic> json) {
    return MiscellaneousInfo(
      id: json['id'] as String?,
      createdBy: json['created_by'] as String?,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : null,
      updatedBy: json['updated_by'] as String?,
      updatedAt: json['updated_at'] != null
          ? DateTime.parse(json['updated_at'] as String)
          : null,
      commentedBy: json['commented_by'] as String?,
      comment: json['comment'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'created_by': createdBy,
      'created_at': createdAt?.toIso8601String(),
      'updated_by': updatedBy,
      'updated_at': updatedAt?.toIso8601String(),
      'commented_by': commentedBy,
      'comment': comment,
    };
  }
}
